cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseConsumerState.java

```diff
@@ -15,30 +15,55 @@
  */
 package co.cask.cdap.data2.transaction.queue.hbase;
 
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.data2.queue.ConsumerConfig;
+import com.google.common.base.Objects;
+
+import javax.annotation.Nullable;
+
 /**
- * Represents state of queue consumer.
+ * Represents state of a queue consumer.
  */
 public final class HBaseConsumerState {
 
-  private final long groupId;
-  private final int instanceId;
+  private final ConsumerConfig consumerConfig;
   private final byte[] startRow;
+  private final byte[] barrierStartRow;
+  private final byte[] barrierEndRow;
 
-  HBaseConsumerState(byte[] startRow, long groupId, int instanceId) {
+  HBaseConsumerState(ConsumerConfig consumerConfig, byte[] startRow,
+                     @Nullable byte[] barrierStartRow, @Nullable byte[] barrierEndRow) {
+    this.consumerConfig = consumerConfig;
     this.startRow = startRow;
-    this.groupId = groupId;
-    this.instanceId = instanceId;
+    this.barrierStartRow = barrierStartRow;
+    this.barrierEndRow = barrierEndRow;
```

Comment:
hmm. Naming - although I think the word barrier is appropriate - is a little strange. If I understand correctly, barrierEndRow is where the config changed, and the consumer waits for all other consumers to each this row. So I would call this simply "barrier". But what is barrierStartRow? Isn't that the barrier of the previous config change? So maybe call it previousBarrier?